components/TextControls/src/BaseTextAreas/private/MDCBaseTextAreaLayout.h

```diff
@@ -0,0 +1,56 @@
+// Copyright 2020-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <UIKit/UIKit.h>
+
+#import "MDCTextControl.h"
+#import "MDCTextControlAssistiveLabelView.h"
+
+@interface MDCBaseTextAreaLayout : NSObject
```

Comment:
Whats the test plan here. I know this object is intending to be a giant function with 10+ inputs and 10+ struct outputs. How do I know that I'm not breaking things by tweaking one of the internal algorithms?

btw lets add a class comment explain the purpose of this object. Something like:
This layout object will calculate the frames and positions of all of the TextArea elements every frame. The inputs are gathered in the initialization of the object and output them as properties to be used in that same layout.

Reply:
I think the best way to test the layout object is through snapshot tests of the text areas themselves. If one of the outputs of the layout object (like `floatingLabelFrame`, for example) changes, that will be very easy to capture in a snapshot test.

Reply:
I added some documentation above the initializer 👍

Reply:
Cool I am expecting a robust set of snapshots for this component.